testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/ComponentsTest.java

```diff
@@ -85,14 +85,16 @@ private void testConcurrency(BiConsumer<ExecutorService, Integer> taskCreator) t
     public void testConcurrencyWithoutChildren() throws InterruptedException {
         testConcurrency((s, i) -> s.submit(new CreateAndDeleteComponent(s, i)));
 
-        assertThat(realm.components().query(realm.toRepresentation().getId(), TestProvider.class.getName()), Matchers.hasSize(0));
+//        the purpose of this test is to test whether it finishes in time, consistency should be resolved by KEYCLOAK-XXXXX
+//        assertThat(realm.components().query(realm.toRepresentation().getId(), TestProvider.class.getName()), Matchers.hasSize(0));
```

Comment:
I don't understand this comment. Could you explain please?

Reply:
The comment will need improvement, there is not guaranteed that concurrent approach to components (and also to other entities, but we do not test other entities) in map store. The reason is that when we start several keycloak transactions at the same time there is race condition. In JPA database is taking care of it. 

I've consulted this test with @hmlnarik as he was the original author. The test was added mainly because there was slowness when components was added back then. We've agreed that the test should stay there to have test that the component creation finishes in certain amount of time. The data consistency should not be tested atm. and the work needed should be tracked by jira (I have create it and put the link there).

Reply:
> Isn't this the same problem as I experienced with the `ConcurrentLoginTest`? Meaning the current map store isn't designed to handle concurrency well. In that case the failing tests should be skipped.

The test was not skipped (reason above), but maybe for map store we can skip `assertThat(realm.components().query(realm.toRepresentation().getId(), TestProvider.class.getName()), Matchers.hasSize(0));` instead of commenting it.